File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int d = -1; 
        for(int i = 0; i < word.length(); i++){
            if(word[i] == ch){
                d = i;
                break; 
            }
        }

        if(d != -1){
            string sub = word.substr(0, d + 1);
            std::reverse(sub.begin(), sub.end());
            string result = sub + word.substr(d + 1); 
            return result;
        }
        
        return word;
    }
};

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:

        if ch not in word:
            return word
            
        ind = word.index(ch)+1

        res = word[:ind]
        ans = word[ind:]

        res = res[::-1]
        return res + ans
        